src/renderer/features/player/hooks/use-scrobble.ts

```diff
@@ -315,12 +315,29 @@ export const useScrobble = () => {
 
     useEffect(() => {
         const unsubSongChange = usePlayerStore.subscribe(
-            (state) => [state.current.song, state.current.time],
+            (state) => [
+                state.current.song,
+                state.current.time,
+                state.current.player,
+                state.current.index,
+            ],
             handleScrobbleFromSongChange,
             {
                 // We need the current time to check the scrobble condition, but we only want to
                 // trigger the callback when the song changes
-                equalityFn: (a, b) => (a[0] as QueueSong)?.id === (b[0] as QueueSong)?.id,
+                // There are three conditions where this should trigger:
+                // 1. The song actually changes (the common case)
+                // 2. The song does not change, but the player dows. This would either be
+                //    a single track on repeat one, or one track added to the queue
+                //    multiple times in a row and playback goes normally (no next/previous)
+                // 3. The song does not change, but the queue position does. This would happen if
+                //    the same song has been enqueued multiple times in a row, and the prev/next
+                //    button is pressed. In this case, the player is forced to 1, which may not cause
+                //    condition 2 to fire. Thus, comparing the queue index will catch this case
```

Comment:
I think the better fix for condition 3 is changing the equalityFn to check the QueueSong by `uniqueId` instead of `id`. Each song added to the queue has a unique identifier that's appended onto the Song object so we wouldn't have to check if the index changed or not.